SEED-platform/seed
# refactor(bsync): fully integrate into import flow

#### Any background context you want to provide?
This pr further integrates BuildingSync import with the regular import flow.
#### What's this PR do?
- parses buildingsync files immediately during 'saving the raw data' task, rather than waiting until the 'mapping' task
- removes the xml specific mapping task, now uses the map_row_chunk
- removes mapping_xml partial for showing the bsync mapping page, now uses the regular mapping partial

Previously, importing BuildingSync files would fail if there were any 'required' fields it found missing. This PR makes a change that ignores any fields it fails to parse for the PropertyState during the import flow. It is now the data quality checker's job to warn users when their properties are missing info (address line 1, PM property id, etc). This change _does_ warn users about bad scenarios/measures/meter data, which is provided after the last step of import ("save mapping") in the data import modal.

#### How should this be manually tested?
Import and export a variety of buildingsync files (`seed/building_sync/tests/data`)
- `buildingsync_ATT_export.xml`: import OK (it should give some warnings about scenarios not having names -- this is due to the way ATT uses Scenarios); should have scenarios and property info
- `buildingsync_ex01_measures_bad_names.xml`: import should be normal until you click "save mappings". It should give a warning that some measure names were bad and some scenarios were not linked to measures as a result.
- `buildingsync_v2_0_bricr_workflow.xml`: import OK; should have scenarios, measures, and Meter data
- `ex_1_no_schemaLocation.xml`: should immediately fail when you try to import it with an error about the schemaLocation.
- `ex_1_and_buildingsync_ex01_measures.zip`: this should show TWO properties on the mapping page. One should show measures and scenario info on the property detail page, while the other should not. Verify that "export as buildingsync" works for each property (e.g. just check it does/doesn't have the measures and scenarios)
